internal/rest/mgmt/helpers.go

```diff
@@ -0,0 +1,83 @@
+/*
```

Comment:
These are the kinds of functions I'd like to see moved into the Go SDK and a proper wrapper developed.

However, they would need to be changed:

-  to take in an external `context` as that is how external developers control the lifecycle of long-running asynchronous tasks. Using context.Background() is an orange flag for a public externally focused function
- would need to work with the API abstractions there instead of directly using the `edge-api` repo. This code does not support HA failover (like all of our CLI tooling at the moment).

Reply:
that's why i put it into the internal folder... i am always cautious with my new, external functions since you can't ever take them back without deprecating etc.

anything "HA" would be spectacular to add here. I'd like to get "rev 1" out first without HA support but if there are other checks that make sense, I would expect this code to expand to help people with that too